bin/reth/src/test_eth_chain/runner.rs

```diff
@@ -1,13 +1,161 @@
+    // It is expected to not execute these tests.
+    if path.file_name() == Some(OsStr::new("accessListExample.json")) ||
+        path.file_name() == Some(OsStr::new("basefeeExample.json")) ||
+        path.file_name() == Some(OsStr::new("eip1559.json")) ||
+        path.file_name() == Some(OsStr::new("mergeTest.json"))
+    {
+        return true
+    }
+
+    // These tests are passing, but they take a lot of time to execute so we are going to skip them.
+    if path.file_name() == Some(OsStr::new("loopExp.json")) ||
+        path.file_name() == Some(OsStr::new("Call50000_sha256.json")) ||
+        path.file_name() == Some(OsStr::new("static_Call50000_sha256.json")) ||
+        path.file_name() == Some(OsStr::new("loopMul.json")) ||
+        path.file_name() == Some(OsStr::new("CALLBlake2f_MaxRounds.json"))
+    {
+        return true
+    }
+    false
+}
 
 /// Run one JSON-encoded Ethereum blockchain test at the specified path.
-pub async fn run_test(path: &Path) -> eyre::Result<()> {
+pub async fn run_test(path: PathBuf) -> eyre::Result<()> {
+    let path = path.as_path();
     let json_file = std::fs::read(path)?;
-    let suits: Test = serde_json::from_reader(&*json_file)?;
+    let suites: Test = serde_json::from_reader(&*json_file)?;
+
+    if should_skip(path) {
+        return Ok(())
+    }
+
+    for (name, suite) in suites.0 {
+        if matches!(
+            suite.network,
+            ForkSpec::ByzantiumToConstantinopleAt5 |
+                ForkSpec::Constantinople |
+                ForkSpec::MergeEOF |
+                ForkSpec::MergeMeterInitCode
+        ) {
+            continue
+        }
+
+        if matches!(suite.pre, State(RootOrState::Root(_))) {}
+
+        let pre_state = match suite.pre.0 {
+            RootOrState::State(state) => state,
+            RootOrState::Root(_) => {
+                info!("Skipping test {name}...");
+                continue
+            }
+        };
+
+        debug!("Executing test: {name} for spec: {:?}", suite.network);
+
+        let spec_upgrades: SpecUpgrades = suite.network.into();
+        // if paris aka merge is not activated we dont have block rewards;
+        let has_block_reward = spec_upgrades.paris != 0;
+
+        // Create db and acquire transaction
+        let db = create_test_rw_db::<WriteMap>();
+        let tx = db.tx_mut()?;
+
+        // insert genesis
+        let header: SealedHeader = suite.genesis_block_header.into();
+        let genesis_block = BlockLocked { header, body: vec![], ommers: vec![] };
+        reth_provider::insert_canonical_block(&tx, &genesis_block, has_block_reward)?;
+
+        suite.blocks.iter().try_for_each(|block| -> eyre::Result<()> {
+            let decoded = BlockLocked::decode(&mut block.rlp.as_ref())?;
+            reth_provider::insert_canonical_block(&tx, &decoded, has_block_reward)?;
+            Ok(())
+        })?;
+
+        pre_state.into_iter().try_for_each(|(address, account)| -> eyre::Result<()> {
```

Comment:
Same as above, adding everything in parallel would be nice
```suggestion
        pre_state.into_par_iter().try_for_each(|(address, account)| -> eyre::Result<()> {
```

Reply:
It needs rw tx, so it will be blocked if it is in parallel.